File: contrib/liio/image_test2.c
#include "iio.h"
int main()
{
	int w, h, pd;
	float *y = iio_read_image_float_vec("/tmp/n.png", &w, &h, &pd);
	for (int j = 0; j < h; j++)
	for (int i = 0; i < w-1; i++)
	for (int l = 0; l < pd; l++)
	{
		int idx = ( j*w + i ) * pd + l;
		y[idx] = y[idx+pd] - y[idx];
	}
	iio_save_image_float_vec("ckuza.tiff", y, w, h, pd);
	return 0;
}

File: contrib/lua/image_test2.lua
local iio = require("imgiio")

local y = iio.read("/tmp/n.png")

local t = os.clock()
print("TIME")

-- the algorithm
y.boundary = 1
for j = 0, y.h - 1 do
for i = 0, y.w - 2 do
for l = 0, y.pd - 1 do
	idx = (j * y.w + i) * y.pd + l
	--y.x[idx] = y(i+1,j,l) - y(i,j,l)
	y.x[idx] = y.x[idx + y.pd] - y.x[idx];
end
end
end

print(os.clock() - t)

iio.write("kuza.tiff", y)
